### PR TITLE
Enable inserting custom URL  using basic auth

### PR DESCRIPTION
**Pull request by tirongi**
In the credential nodeId can be inserted a custom server URL and the basic auth will be used.

### CODE REVIEW

File: packages/components/credentials/ElectricsearchUserPassword.credential.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ class ElasticSearchUserPassword implements INodeCredential {
             'Refer to <a target="_blank" href="https://www.elastic.co/guide/en/kibana/current/tutorial-secure-access-to-kibana.html">official guide</a> on how to get User Password from ElasticSearch'
         this.inputs = [
             {
-                label: 'Cloud ID',
+                label: 'Cloud ID or custom server URL',
                 name: 'cloudId',
```
Comment (HenryHengZJ) [inline packages/components/credentials/ElectricsearchUserPassword.credential.ts:19]: can we add a `description` to explain how to get Cloud ID OR what is custom server URL? Better if it has link to docs like [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/credentials/AWSCredential.credential.ts#L16)

File: packages/components/nodes/vectorstores/Elasticsearch/ElasticSearchBase.ts [hunk 2]
```diff
@@ -144,15 +144,30 @@ export abstract class ElasticSearchBase {
...
             let password = getCredentialParam('password', credentialData, nodeData)
-            elasticSearchClientOptions = {
-                cloud: {
-                    id: cloudId
-                },
-                auth: {
-                    username: username,
-                    password: password
+            if (cloudId.startsWith('http')) {
+                let username = getCredentialParam('username', credentialData, nodeData)
```
Comment (HenryHengZJ) [inline packages/components/nodes/vectorstores/Elasticsearch/ElasticSearchBase.ts:148]: these 2 lines are redundant as its already declared above

**Comment by tirongi** [comment]
Thanks for your suggestions.
I just added the explanation, you can use an Elastic Cloud ID or put the URL address to the Elastic server instance in which the index is stored.
Based on the given string the client will be configured with the right auth settings. 
See: [Elastic guide](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-connecting.html)

**Review by HenryHengZJ — APPROVED** [review]
thanks @tirongi !

**Pull request merged**